### PR TITLE
CI: skip appveyor for doc only change

### PR DESCRIPTION
**Pull request by rcomer**
## PR Summary

Consistent with recent changes for CI on other platforms (#25145, #25261), this change should mean the Appveyor CI is skipped if the only changes are under `doc` or in the `galleries`, or if `[ci doc]` is in the commit message.  Following the guidance here:
https://www.appveyor.com/docs/how-to/filtering-commits/

I added `[ci doc]` to the current commit, to test it works.

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -9,6 +9,14 @@ branches:
     - /auto-backport-.*/
     - /^v\d+\.\d+\.[\dx]+-doc$/
 
+skip_commits:
+  message: /\[ci doc\]/
+  files:
+    - 'doc/**'
+    - 'galleries/examples/**'
+    - 'galleries/plot_types/**'
+    - 'galleries/tutorials/**'
```
Comment (greglucas) [inline .appveyor.yml:18]:
```suggestion
    - 'galleries/**'
```
I think we can just skip the entire galleries directory.

**Review by greglucas — APPROVED** [review]
Looks good, but oddly enough this worked while the GitHub Actions tests still ran! So the GitHub Actions version might be incorrect...

**Comment by rcomer** [comment]
> the GitHub Actions tests still ran!

Yeah, I've been digging around the docs for that, as I've noticed it before.  From what I can tell, `head_commit` is a property of a [push event](https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads#push), but not of a [pull_request event](https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request).  So possibly it just can't work for PR CI?

**Review by ksunden — APPROVED** [review]

**Comment by jklymak** [comment]
@rcomer maybe self merge if you are done?

**Pull request merged**